### PR TITLE
[update] ppp_device support private APN access point

### PR DESCRIPTION
**Pull request by xiangxistu**
* ppp_device support private APN access point
* supprot cmux function
Signed-off-by: liuxinaliang <liuxianliang@rt-thread.com>

### CODE REVIEW

File: iot/ppp_device/Kconfig [hunk 1]
```diff
@@ -7,6 +7,7 @@ menuconfig PKG_USING_PPP_DEVICE
     select RT_USING_LWIP
     select RT_LWIP_PPP
     select RT_LWIP_PPPOS
+    select CMUX_USING_GSM
```
Comment (armink) [inline iot/ppp_device/Kconfig:10]: 这块由用户来决定吧，先不要强制依赖了

File: iot/ppp_device/Kconfig [hunk 2]
```diff
@@ -50,6 +51,17 @@ if PKG_USING_PPP_DEVICE
 
     endif
 
+    choice
+    prompt "APN access point choice"
+    default USING_PUBLIC_APN
```
Comment (armink) [inline iot/ppp_device/Kconfig:56]: 这些宏注意开头的前缀

File: iot/ppp_device/Kconfig [hunk 3]
```diff
@@ -62,8 +74,20 @@ if PKG_USING_PPP_DEVICE
...
+    endchoice
+    endif
 
-        endchoice
+    if USING_PRIVATE_APN
+    config PPP_PRIVATE_APN
+        string "the name of APN access point"
+        default "xxx"
+
+    config PRIVATE_APN_ACCOUNT
```
Comment (armink) [inline iot/ppp_device/Kconfig:85]: 同上

File: iot/ppp_device/Kconfig (hunk 1 above)
Comment (xiangxistu) [reply iot/ppp_device/Kconfig]: 已经取消依赖

File: iot/ppp_device/Kconfig (hunk 3 above)
Comment (xiangxistu) [reply iot/ppp_device/Kconfig]: 源码中也已经更改

**Pull request merged**